src/test/java/org/folio/search/controller/BrowseCallNumberIrrelevantResultTest.java

```diff
@@ -0,0 +1,196 @@
+import static org.folio.search.utils.TestUtils.randomId;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import org.folio.search.domain.dto.CallNumberBrowseResult;
+import org.folio.search.domain.dto.Holding;
+import org.folio.search.domain.dto.Instance;
+import org.folio.search.domain.dto.Item;
+import org.folio.search.domain.dto.ItemEffectiveCallNumberComponents;
+import org.folio.search.domain.dto.Metadata;
+import org.folio.search.domain.dto.Tags;
+import org.folio.search.support.base.BaseIntegrationTest;
+import org.folio.search.utils.CallNumberUtils;
+import org.folio.spring.test.type.IntegrationTest;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+@IntegrationTest
+class BrowseCallNumberIrrelevantResultTest extends BaseIntegrationTest {
+
+  private static final Instance[] INSTANCES = instances();
+  private static final Map<String, Instance> INSTANCE_MAP =
+    Arrays.stream(INSTANCES).collect(toMap(Instance::getTitle, identity()));
+
+  @BeforeAll
+  static void prepare() {
+    setUpTenant(INSTANCES);
+  }
+
+  @AfterAll
+  static void cleanUp() {
+    removeTenant();
+  }
+
+  @Test
+  void browseByCallNumber_browsingAroundWithEnabledIntermediateValues() {
+    enableFeature(BROWSE_CN_INTERMEDIATE_VALUES);
+
+    var request = get(instanceCallNumberBrowsePath())
+      .param("callNumberType", "dewey")
+      .param("query", prepareQuery("typedCallNumber >= {value} or typedCallNumber < {value}", "308 H977"))
+      .param("limit", "15")
+      .param("highlightMatch", "true")
+      .param("precedingRecordsCount", "5")
+      .param("expandAll", "true");
+    var actual = parseResponse(doGet(request), CallNumberBrowseResult.class);
+    var expected = cnBrowseResult(2, List.of(
+      cnBrowseItem(instance("instance #01"), "308 H977", true)
+    ));
+    expected.setItems(CallNumberUtils.excludeIrrelevantResultItems("dewey", expected.getItems()));
+    assertThat(actual).isEqualTo(expected);
+
+    disableFeature(BROWSE_CN_INTERMEDIATE_VALUES);
+  }
+
+  @Test
+  void browseByCallNumber_browsingAroundWithDisabledIntermediateValuesAndWithoutTypeAndLowLimit() {
+    var limit = 2;
+    var request = get(instanceCallNumberBrowsePath())
+      .param("query", prepareQuery("callNumber >= {value} or callNumber < {value}", "308 H977"))
+      .param("limit", String.valueOf(limit))
+      .param("highlightMatch", "true")
+      .param("precedingRecordsCount", "1")
+      .param("expandAll", "true");
+    var actual = parseResponse(doGet(request), CallNumberBrowseResult.class);
+    var expected = cnBrowseResult(3, List.of(
+      cnBrowseItem(instance("instance #01"), "308 H977", true)
+    )).next("3308 H977");
+    assertThat(actual).isEqualTo(expected);
+    assertThat(actual.getItems()).hasSizeLessThanOrEqualTo(limit);
+  }
```

Comment:
There is no sense in the test as it works with `callNumber` and not `typedCallNumber` .
You need to create a test where based on query elastic search could return more than a limit. With the test we should understand that pagination works fine and we get full page even with you filtration.